sklearn/preprocessing/_encoders.py

```diff
@@ -786,7 +786,13 @@ def fit(self, X, y=None):
                             f"handle_unknown is 'use_encoded_value', "
                             f"got {self.unknown_value}.")
 
-        self._fit(X, force_all_finite='allow-nan')
+        if self.handle_unknown == 'error':
+            handle_unknown = 'error'
+        else:
+            handle_unknown = 'ignore'
+
+        self._fit(X, handle_unknown=handle_unknown,
+                  force_all_finite='allow-nan')
```

Comment:
I guess you could also just pass `self.handle_unknown`, since as long as it's not `"error"` it'd work. `_fit` doesn't do validation on the value of `handle_unknown`. I'm not sure though, WDYT @jnothman

Reply:
I would prefer that we pass only "error" or "ignore". It is less surprising. Otherwise, we would need to add a comment.

Reply:
Actually, it could make sense to add a comment now as well :)

Reply:
And looking closer at the code, I agree with @adrinjalali that it would be fine to pass directly the string.
I added a comment on the top however.